src/webgpu/shader/execution/memory_model/barrier.spec.ts

```diff
@@ -216,11 +271,21 @@ g.test('workgroup_barrier_load_store')
       t.selectDeviceOrSkipTestCase('shader-f16');
     }
     t.skipIf(
-      !t.params.normalBarrier && t.params.memType === MemoryType.NonAtomicStorageClass,
+      !t.params.normalBarrier && t.params.memType !== MemoryType.NonAtomicWorkgroupClass,
       'workgroupUniformLoad does not have storage memory semantics'
     );
+    t.skipIf(
+      t.params.memType === MemoryType.NonAtomicTextureClass && t.params.accessValueType !== 'u32',
```

Comment:
nit: I'd prefer to `t.params.accessValueType === 'f16'` here.
That's more direct logic, and resilient to adding other access types like f32 or i64